src/process.rs

```diff
@@ -237,28 +237,23 @@ define_id_getter! {Process}
 
 fn read_process_availabilities_from_iter<I>(
     iter: I,
-    file_path: &Path,
     process_ids: &HashSet<Rc<str>>,
     time_slice_info: &TimeSliceInfo,
 ) -> Result<HashMap<Rc<str>, Vec<ProcessAvailability>>>
 where
     I: Iterator<Item = ProcessAvailabilityRaw>,
 {
-    let availabilities = iter
-        .map(|record| {
-            let time_slice = time_slice_info
-                .get_selection(&record.time_slice)
-                .unwrap_input_err(file_path);
-
-            ProcessAvailability {
-                process_id: record.process_id,
-                limit_type: record.limit_type,
-                time_slice,
-                value: record.value,
-            }
-        })
-        .into_id_map(process_ids)
-        .unwrap_input_err(file_path);
+    let mut records: Vec<ProcessAvailability> = Vec::new();
+    for record in iter {
+        let time_slice = time_slice_info.get_selection(&record.time_slice)?;
+        records.push(ProcessAvailability {
+            process_id: record.process_id,
+            limit_type: record.limit_type,
+            time_slice,
+            value: record.value,
+        });
+    }
+    let availabilities = records.into_iter().into_id_map(process_ids)?;
```

Comment:
Actually, on second thoughts, you can keep the use of `map()` here and avoid creating the `Vec`:

```suggestion
    let availabilities = iter
        .map(|record| -> Result<_> {
            let time_slice = time_slice_info.get_selection(&record.time_slice)?;

            Ok(ProcessAvailability {
                process_id: record.process_id,
                limit_type: record.limit_type,
                time_slice,
                value: record.value,
            })
        })
        .process_results(|iter| iter.into_id_map(process_ids))??;
```